src/pages/Fox/hooks/useFoxWifHatMerkleTreeQuery.ts

```diff
@@ -0,0 +1,50 @@
+import { useQuery } from '@tanstack/react-query'
+import axios from 'axios'
+import { useMemo } from 'react'
+import { IPFS_GATEWAY } from 'pages/RFOX/constants'
+
+export type FoxWifHatClaimQuote = { index: bigint; amount: string; proof: `0x${string}`[] }
+
+type MerkleData = {
+  merkleRoot: string
+  tokenTotal: string
+  claims: Record<string, FoxWifHatClaimQuote>
+}
+
+type FoxWifHatMerkleTreeQueryKey = ['foxWifHatMerkleTree']
+
+const MERKLE_TREE_HASH = 'bafybeifaysj3qdmwrjlkc7pebbutjj7qdtyfofxxqdgyar2xzyrqy3d2ui'
+
+export const getFoxWifHatMerkleTreeQueryKey = (): FoxWifHatMerkleTreeQueryKey => [
+  'foxWifHatMerkleTree',
+]
+
+export const fetchFoxWifhatMerkleTree = async (): Promise<MerkleData> => {
+  const { data } = await axios.get<MerkleData>(`${IPFS_GATEWAY}/${MERKLE_TREE_HASH}`)
+
+  const lowercasedClaims = Object.entries(data.claims).reduce(
+    (acc, [address, claim]) => {
+      acc[address.toLowerCase()] = claim
+      return acc
+    },
+    {} as typeof data.claims,
+  )
+
+  return {
+    ...data,
+    claims: lowercasedClaims,
+  }
+}
```

Comment:
preferably-blocking: *fetching* all claims is fine, but can we use `select` to only keep the ones for the current accountIds set? 

https://jam.dev/c/f23fb73a-54f8-4550-8731-8b1c919cb357

Reply:
Also, a pattern we can use re: `select` here is to have a default select *and/or* pass a `select` override from props.
Noticing quite a few occurences of `claims[fromAccountId(accountId).account` across the board, so could potentially use a single `queryFn` (no accounts in `queryKey`, since the claims query fetches all anyway), and then narrow down by either portfolio AccountIds array (when displaying in a list), *or* an array of a single current accountId.

This way, no need to lookup across the board and we can keep things a bit cleaner and DRYer.

Reply:
Done in [7e9a198](https://github.com/shapeshift/web/pull/8611/commits/7e9a198a552c9ffbe24675fdb83a91d1875e342b)